boards: dts: Add i2c to nrf5X_pcaX board dts.

hakonfam opened the pull request:
Fixes #7137

Enables the i2c drivers for these boards, allowing I2C to be used.
Required to enable i2c with these boards.
Tested building samples/bluetooth/beacon for nrf52_pca10040 with i2c enabled.

Parameters used for enabling i2c:
CONFIG_I2C=y
CONFIG_I2C_0=y
CONFIG_I2C_NRF5=y
CONFIG_I2C_NRF5_0_GPIO_SCL_PIN=6
CONFIG_I2C_NRF5_0_GPIO_SDA_PIN=7

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.com>

carlescufi reviewed (changes requested):
Why not add this to the 52840_pca10056 as well?

Commit:
boards: dts: Add i2c to nrf5X_pcaX board dts.

Fixes #7137

Enables the i2c drivers for these boards.
Required to enable i2c with these boards.
Tested samples/bluetooth/beacon for nrf52_pca10040
with i2c enabled.

Parameters used for enabling i2c:
CONFIG_I2C=y
CONFIG_I2C_0=y
CONFIG_I2C_NRF5=y
CONFIG_I2C_NRF5_0_GPIO_SCL_PIN=6
CONFIG_I2C_NRF5_0_GPIO_SDA_PIN=7

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.com>

hakonfam commented:
Added to nrf52840_pca10056 as well.

carlescufi reviewed (approved):
Approved, but you'll need to fix your commit message. One of the lines is too long.

https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/13694/5/tests
